pkg/clients/clients.go

```diff
@@ -161,25 +167,52 @@ func (dcs *DynamicClientSet) gvkForKind(kind Kind) (gvk schema.GroupVersionKind,
 	return
 }
 
-func (dcs *DynamicClientSet) namespaced(gvk schema.GroupVersionKind) bool {
-	resources, err := dcs.DiscoveryClientCached.ServerPreferredResources()
+func (dcs *DynamicClientSet) namespaced(gvk schema.GroupVersionKind) (bool, error) {
+	// Handle known non-namespaced Kinds.
+	switch Kind(gvk.Kind) {
+	case APIService, CertificateSigningRequest, ClusterRole, ClusterRoleBinding, CustomResourceDefinition,
+		MutatingWebhookConfiguration, Namespace, PersistentVolume, PodSecurityPolicy, PriorityClass,
+		StorageClass, ValidatingWebhookConfiguration:
+		return false, nil
+	}
+
+	resources, err := dcs.getServerResources()
 	if err != nil {
-		// Default to true
-		return true
+		return false, err
 	}
 
 	for _, gvResources := range resources {
 		if gvResources.GroupVersion == gvk.GroupVersion().String() {
 			for _, resource := range gvResources.APIResources {
 				if resource.Kind == gvk.Kind {
-					return resource.Namespaced
+					return resource.Namespaced, nil
 				}
 			}
 		}
 	}
 
-	// Default to true
-	return true
+	return false, fmt.Errorf("failed to discover namespace info for %s", gvk)
+}
+
+func (dcs *DynamicClientSet) getServerResources() ([]*v1.APIResourceList, error) {
+	var resources []*v1.APIResourceList
+	err := retry.SleepingRetry(
```

Comment:
Combining multiple exponential backoff conditions might cause some of our timeouts to fail more frequently. We've tried to put the retry logic in one place to make it really predictable. I'm fine shipping this for now, but highly recommend we come back later and audit the retry path and make sure it actually _is_ predictable.